internal/pkg/template/diff/write_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Integration_Parse_Write(t *testing.T) {
+	testCases := map[string]struct {
+		curr        string
+		old         string
+		wanted      string
+		wantedError error
+	}{
+		"add a map": {
+			curr: `Mary:
+  Height:
+    cm: 168
+  Weight:
+    kg: 52`,
+			old: `Mary:
+  Height:
+    cm: 168`,
+
+			wanted: `~ Mary:
+    + Weight:
+    +     kg: 52
+`,
+		},
+		"remove a map": {
+			curr: `Mary:
+  Height:
+    cm: 168`,
+			old: `Mary:
+  Height:
+    cm: 168
+  Weight:
+    kg: 52`,
+			wanted: `~ Mary:
+    - Weight:
+    -     kg: 52
+`,
+		},
+		"change keyed values": {
+			curr: `Mary:
+  Height:
+    cm: 168`,
+			old: `Mary:
+  Height:
+    cm: 190`,
+			wanted: `~ Mary:
+    ~ Height:
+        ~ cm: 190 -> 168
+`,
+		},
+		"list does not change": {
+			old:    `Alphabet: [a,b,c,d]`,
+			curr:   `Alphabet: [a,b,c,d]`,
+			wanted: "No changes.\n",
+		},
+		"list reordered": {
+			// TODO(lou1425926): complete the test.
```

Comment:
Isn't this testcase captured by "list with a scalar value changed" below?

Reply:
yeahhh I left it here just in case I have the time to implement the logic that writes ⬇️  😂 I think of this as a nice to have though 😂 
```
bear(moved from 2 to 3), dog (moved from 3 to 2)
```

(not important but instead of "scalar value changed", it'd actually be an insertion (at the new position)  + a deletion (from the old position))